packages/svelte-ux/src/lib/utils/format.ts

```diff
@@ -1,24 +1,41 @@
 import { isFunction } from 'lodash-es';
 
 import { formatDate, PeriodType } from './date';
-import { formatNumberAsStyle } from './number';
-import type { FormatNumberStyle } from './number';
+import { formatNumber } from './number';
+import type { FormatNumberOptions, FormatNumberStyle } from './number';
 
-export type FormatType = FormatNumberStyle | PeriodType | ((value: any, ...extraArgs) => any);
+export type FormatType =
+  | FormatNumberStyle
+  | PeriodType
+  | ((value: any, ...extraArgs: any[]) => any);
 
 /**
  * Generic format which can handle Dates, Numbers, or custom format function
  */
-export function format(value: any, format?: FormatType, ...extraFuncArgs: any[]) {
+export function format(
+  value: number | null,
+  format?: FormatNumberStyle,
+  extraFuncArgs?: FormatNumberOptions
+): string;
+export function format(
+  value: string | Date | null | undefined,
+  format?: PeriodType,
+  ...extraFuncArgs: any[]
+): string;
+export function format(value: any, format?: FormatType, ...extraFuncArgs: any[]): any {
   let formattedValue = value ?? ''; // Do not render `null`
 
   if (format) {
+    // TODO QUESTION: is this used?
```

Comment:
It is.  It's helpful for `<Table>` column formats when you want to do something special, such as call another format function like `humanizeDuration()` or any other formatting like only show the last X digits of an ID (SSN, SHA, etc)

```svelte
<Table columns={[
  // ...
  {
    name: 'startDate',
    format: PeriodType.Day,
  },
  {
    name: 'cost',
    format: 'currency'
  },
  {
    name: 'duration',
    format: value => humanizeDuration({ start: value });
  }
]} />
```

The same is useful in LayerChart to format Axis ticks (`<Axis format={value => ...} />`), and there are other examples.